### ISSUE TITLE
default core clock limit hard wall

### ISSUE THREAD

**Issue by GloriousEggroll** [BoukeHaarsma23/WattmanGTK#8]
hi, i was hitting an issue with my vega 64 with hitting a 1630 default soft wall limit on the core. I found via the arch wiki:
https://wiki.archlinux.org/index.php/AMDGPU#Overclocking
you can increase the limit. I set
echo "10" > /sys/class/drm/card0/device/pp_sclk_od
to allow a 10% max increase from default, and now i am able to get past the soft wall

**Comment by Haxk20** [comment]
Try echo "high" > /sys/class/drm/card0/device/power_dpm_force_performance_level as root.
This will put the card in highest core and memory speeds.
I had similar issue with RX560X where auto performance level put the card in lowest clocks. Switching it to high allowed me to do so much more.
Also disable that overclock since if you put the card in high and with 10% overclock this might damage the card completely.

**Comment by Haxk20** [comment]
This should be reported to AMDGPU buglist as im experiencing the same issue. But on RX560X.
Sadly i have to set high performance mode on my card because i cant get past the soft wall on memory clocks. Sadly that way my card runs 100% all the time while using it for gaming.

**Comment by BoukeHaarsma23** [comment]
What do you mean by "soft wall"? The problem is a bit unclear to me

**Comment by Haxk20** [comment]
> What do you mean by "soft wall"? The problem is a bit unclear to me

GPU clocks are at the lowest state until you "overclock" the GPU then it starts to change states depending on load. Without overclocking it or chaging performance state its stuck at lowest clock state.

**Comment by GloriousEggroll** [comment]
the vega 64 is able to  clock much higher than 1630, but by default on linux it's limited to 1630 no matter what you set unless you increase the max limit first as mentioned above

@Haxk20 my gpu goes through all the clock states properly. the max advertised boost on my card is 1590. the problem is it can normally go much higher than that. I was hitting a wall at 1630. Once i tried to set any state higher than 1630, it would default to  the previous state. Once I added the fix mentioned above, it allowed clocking higher than 1630. 

To put this in perspective. some people have managed to clock their vega 64s on average 1645-1680 safely, and I had done this on windows.

**Comment by Haxk20** [comment]
Weird because when i do your "fix" then my states start working properly. But this really isnt the proper solution as the driver should switch states normally without the need of me switching the first state.

**Comment by BoukeHaarsma23** [comment]
I was able to reproduce this. However, since I will not write any overclock without the user knowledge I updated the FAQ 7bae16e589895437a943fe89cb57f447b415e40e and will close this.

**Comment by GloriousEggroll** [comment]
the option doesn't write an overclock, it only increases the amount the card is able to extend to. it does not change any of the card's default clock speeds

**Comment by BoukeHaarsma23** [comment]
Then I think we misunderstand each other. Or I don't understand everything properly. 

Does
> echo "10" > /sys/class/drm/card0/device/pp_sclk_od
Not overclock your GPU core by 10%?

**Comment by GloriousEggroll** [comment]
no. it increases the max limit that it is able to clock to IF manually set.

think of it like this:
you put the gpu in a box.
you then take the gpu out, and put it in a bigger box, 10% bigger.
the gpu (core) still stays the same size (clock) unless you manually set it to a larger size (core)

**Comment by BoukeHaarsma23** [comment]
I see, I misread the documentation on this. Thanks for pointing this out!

**Comment by GloriousEggroll** [comment]
he used it with the automatic default scaling values. the option should not be set if not setting manual values otherwise the gpu will try to auto-scale overclock to the max value in the range.  So what you would want to do is turn it on only if the user enables manual clock setting mode, then if they disable manual clock mode set it back to 0.

**Comment by BoukeHaarsma23** [comment]
Ah gotcha, working on it ;)
